### PR TITLE
s/slave/replica in guestbook app

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: `s/slave/replica/` in the agnhost guestbook test app.

Redis did this long ago, so can we.
https://redis.io/commands/slaveof
> A note about the word slave used in this man page and command name: Starting with Redis 5 this command: starting with Redis version 5, if not for backward compatibility, the Redis project no longer uses the word slave. Please use the new command REPLICAOF. The command SLAVEOF will continue to work for backward compatibility.

AFAICT guestbook is very roughly trivial-toy-mode-redis for testing purposes.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

We'll need a small follow-up to upgrade the image / swap the flag.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/sig testing
/priority important-soon
/cc @spiffxp @dims @liggitt

### CODE REVIEW

File: test/images/agnhost/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-2.8
+2.9
```
Comment (BenTheElder) [inline test/images/agnhost/VERSION]: maybe we should start just using the sha or something instead of a version number bound to have conflicts ...?

Comment (BenTheElder) [reply test/images/agnhost/VERSION]: discussing in #release-management

**Comment by aojea** [comment]
```
 aojea  Antonios-MacBook-Pro  ~  work  src  k8s.io  kubernetes  master  32⬇  8?  $  grep -ri redis test/ | grep slave
test//cmd/template-output.sh:  output_message=$(kubectl "${kube_flags[@]:?}" expose -f hack/testdata/redis-slave-replicaset.yaml --save-config --port=80 --target-port=8000 --dry-run --template="{{ .metadata.name }}:")
test//cmd/template-output.sh:  kube::test::if_has_string "${output_message}" 'redis-slave:'
test//cmd/apps.sh:  kubectl create -f hack/testdata/redis-slave-replicaset.yaml "${kube_flags[@]:?}"
test//cmd/apps.sh:  # Post-condition: frontend and redis-slave
test//cmd/apps.sh:  kube::test::get_object_assert rs "{{range.items}}{{${id_field:?}}}:{{end}}" 'frontend:redis-slave:'
test//cmd/apps.sh:  # Pre-condition: frontend and redis-slave
test//cmd/apps.sh:  kube::test::get_object_assert rs "{{range.items}}{{${id_field:?}}}:{{end}}" 'frontend:redis-slave:'
test//cmd/apps.sh:  kubectl delete rs frontend redis-slave "${kube_flags[@]:?}" # delete multiple replica sets at once
test//cmd/core.sh:  kubectl create -f test/e2e/testing-manifests/guestbook/redis-slave-service.yaml "${kube_flags[@]}"
test//cmd/core.sh:  # Post-condition: redis-master and redis-slave services are created
test//cmd/core.sh:  kube::test::get_object_assert services "{{range.items}}{{$id_field}}:{{end}}" 'kubernetes:redis-master:redis-slave:'
test//cmd/core.sh:  # Pre-condition: redis-master and redis-slave services exist
test//cmd/core.sh:  kube::test::get_object_assert services "{{range.items}}{{$id_field}}:{{end}}" 'kubernetes:redis-master:redis-slave:'
test//cmd/core.sh:  kubectl delete services redis-master redis-slave "${kube_flags[@]}" # delete multiple services at once
test//cmd/core.sh:  kubectl create -f test/e2e/testing-manifests/guestbook/legacy/redis-slave-controller.yaml "${kube_flags[@]}"
test//cmd/core.sh:  kubectl scale rc/redis-master rc/redis-slave --replicas=4 "${kube_flags[@]}"
test//cmd/core.sh:  kube::test::get_object_assert 'rc redis-slave' "{{$rc_replicas_field}}" '4'
test//cmd/core.sh:  kubectl delete rc redis-{master,slave} "${kube_flags[@]}"
test//cmd/core.sh:  kubectl create -f test/e2e/testing-manifests/guestbook/legacy/redis-slave-controller.yaml "${kube_flags[@]}"
test//cmd/core.sh:  # Post-condition: frontend and redis-slave
test//cmd/core.sh:  kube::test::get_object_assert rc "{{range.items}}{{$id_field}}:{{end}}" 'frontend:redis-slave:'
test//cmd/core.sh:  # Pre-condition: frontend and redis-slave
test//cmd/core.sh:  kube::test::get_object_assert rc "{{range.items}}{{$id_field}}:{{end}}" 'frontend:redis-slave:'
test//cmd/core.sh:  kubectl delete rc frontend redis-slave "${kube_flags[@]}" # delete multiple controllers at once
test//fixtures/doc-yaml/user-guide/replicaset/redis-slave.yaml:  name: redis-slave
test//fixtures/doc-yaml/user-guide/replicaset/redis-slave.yaml:    app: redis
test//fixtures/doc-yaml/user-guide/replicaset/redis-slave.yaml:      app: redis
test//fixtures/doc-yaml/user-guide/replicaset/redis-slave.yaml:        app: redis
test//fixtures/doc-yaml/user-guide/replicaset/redis-slave.yaml:        image: docker.io/library/redis:5.0.5-alpine
test//fixtures/doc-yaml/user-guide/replicaset/redis-slave.yaml:        # https://github.com/kubernetes/examples/blob/97c7ed0eb6555a4b667d2877f965d392e00abc45/guestbook/redis-slave/run.sh
test//fixtures/doc-yaml/user-guide/replicaset/redis-slave.yaml:        command: [ "redis-server", "--slaveof", "redis-master", "6379" ]
test//e2e/apps/deployment.go:   // Create a deployment to delete webserver pods and instead bring up redis-slave pods.
test//e2e/apps/deployment.go:   deploymentName, deploymentImageName := "test-rollover-deployment", "redis-slave"
test//e2e/apps/deployment.go:   deploymentImage := "gcr.io/google_samples/gb-redisslave:nonexistent"
test//e2e/apps/statefulset.go:  return "redis: master/slave"
test//e2e/testing-manifests/guestbook/redis-slave-deployment.yaml.in:  name: redis-slave
test//e2e/testing-manifests/guestbook/redis-slave-deployment.yaml.in:      app: redis
test//e2e/testing-manifests/guestbook/redis-slave-deployment.yaml.in:        app: redis
test//e2e/testing-manifests/guestbook/redis-slave-deployment.yaml.in:        image: {{.RedisImage}}
test//e2e/testing-manifests/guestbook/redis-slave-deployment.yaml.in:        # https://github.com/kubernetes/examples/blob/97c7ed0eb6555a4b667d2877f965d392e00abc45/guestbook/redis-slave/run.sh
test//e2e/testing-manifests/guestbook/redis-slave-deployment.yaml.in:        command: [ "redis-server", "--slaveof", "redis-master", "6379" ]
test//e2e/testing-manifests/guestbook/legacy/redis-slave-controller.yaml:  name: redis-slave
test//e2e/testing-manifests/guestbook/legacy/redis-slave-controller.yaml:    app: redis
test//e2e/testing-manifests/guestbook/legacy/redis-slave-controller.yaml:        app: redis
test//e2e/testing-manifests/guestbook/legacy/redis-slave-controller.yaml:        image: docker.io/library/redis:5.0.5-alpine
test//e2e/testing-manifests/guestbook/legacy/redis-slave-controller.yaml:        # https://github.com/kubernetes/examples/blob/97c7ed0eb6555a4b667d2877f965d392e00abc45/guestbook/redis-slave/run.sh
test//e2e/testing-manifests/guestbook/legacy/redis-slave-controller.yaml:        command: [ "redis-server", "--slaveof", "redis-master", "6379" ]
test//e2e/testing-manifests/guestbook/redis-slave-service.yaml:  name: redis-slave
test//e2e/testing-manifests/guestbook/redis-slave-service.yaml:    app: redis
test//e2e/testing-manifests/guestbook/redis-slave-service.yaml:    app: redis
```

🤷‍♂

**Comment by BenTheElder** [comment]
@aojea the old redis stuff in test/cmd is another issue. We certainly have other dated redis references. That doesn't mean we shouldn't fix them.

The test/e2e/... guestbook manifests specifically will be trivial to update but should be a follow up after the image is (auto?) pushed.

**Comment by aojea** [comment]
:+1:

**Comment by liggitt** [comment]
/lgtm
/approve
/hold for resolution of https://github.com/kubernetes/kubernetes/pull/87105/files#r365534715
(feel free to unhold when ready)

File: test/images/agnhost/VERSION (hunk 1 above)
Comment (BenTheElder) [reply test/images/agnhost/VERSION]: this is probably a good idea to deal with in a follow up, additionally though auto-image pushing currently does not work, so I'll go look into that first...

Comment (BenTheElder) [reply test/images/agnhost/VERSION]: https://github.com/kubernetes/kubernetes/pull/87162

**Comment by BenTheElder** [comment]
I will fix this up next week most likely, was waiting on the image build
fixes and tackling some other infra issues.

**Comment by spiffxp** [comment]
needs rebase

**Comment by BenTheElder** [comment]
rebased

**Comment by BenTheElder** [comment]
ugh
/retest

**Pull request merged**